src/main/java/com/questhelper/helpers/quests/theheartofdarkness/TheHeartOfDarkness.java

```diff
@@ -0,0 +1,1028 @@
+                "of Ascension to the south-east of the salvager overlook again.");
+        talkToPrinceAfterRecruits.addDialogStep("Could you remind me what Ximoua is?");
+        talkToJanus = new NpcStep(this, NpcID.FOREBEARER_JANUS, new WorldPoint(1638, 3224, 0), "Talk to Forebearer Janus inside the tower.");
+
+        // First trial section
+        climbUpToFirstTrial = new ObjectStep(this, ObjectID.STAIRCASE_54369, new WorldPoint(1635, 3221, 0), "Climb up the staircase in the tower to the first" +
+                " challenge.");
+        pickpocketAscended = new NpcStep(this, NpcID.EMISSARY_ASCENDED_13768, new WorldPoint(1641, 3225, 1), "Pickpocket one of the emissary ascended.", true);
+        ((NpcStep) pickpocketAscended).addAlternateNpcs(NpcID.EMISSARY_ASCENDED_13767);
+        useKeyOnSouthEastGate = new ObjectStep(this, NullObjectID.NULL_55354, new WorldPoint(1644, 3220, 1), "Use the key to open the gate in the south-east " +
+                "corner of the room.", towerKey);
+        searchChestForBookAndPaper = new ObjectStep(this, ObjectID.CHEST_54372, new WorldPoint(1644, 3217, 1), "Search the south-east chest for a book and " +
+                "some paper.");
+        openKeywordChestSouthWest = new LockedChestPuzzle(this);
+        readPoem = new DetailedQuestStep(this, "Read the poem.", poem.highlighted());
+        talkToPrinceAfterPoem = new NpcStep(this, NpcID.PRINCE_ITZLA_ARKAN_13770, new WorldPoint(1638, 3218, 1), "Talk to the prince in the room.");
+        talkToPrinceAfterPoem.addDialogSteps("Makt.", "Takam.", "Silam.", "Thanks! Could you translate another?");
+        openKeywordChestNorthWest = new ObjectStep(this, ObjectID.CHEST_54374, new WorldPoint(1636, 3225, 1), "Open the north-west chest.");
+        inputArrows = new ArrowChestPuzzleStep(this);
+        combineScraps = new DetailedQuestStep(this,  "Inspect one of the scraps to combine them.", scrapOfPaper1.highlighted(), scrapOfPaper2.highlighted(),
+                scrapOfPaper3.highlighted());
+
+        readCompletedNote = new DetailedQuestStep(this, "Read the completed note.", completedNote.highlighted());
+        tellJanusPasscode = new NpcStep(this, NpcID.FOREBEARER_JANUS_13766, new WorldPoint(1644, 3226, 1), "Talk to Janus and tell him the passcode.");
+        tellJanusPasscode.addDialogSteps("About that passphrase...", "Yes.");
+
+        // Second trial section
+        startCombatTrial = new NpcStep(this, NpcID.FOREBEARER_JANUS_13766, new WorldPoint(1644, 3225, 2), "Talk to Forebearer Janus, ready to fight.");
+        startCombatTrial.addDialogStep("Yes! I'm ready.");
+        completeCombatTrial = new NpcStep(this, NpcID.EMISSARY_BRAWLER, new WorldPoint(1644, 3225, 2), "Defeat the waves of enemies. Use Protect from " +
+                "Melee. You can heal yourself and the prince using bandages from the chests in the area.", true);
```

Comment:
this step was flickering back to the "startCombatTrial" step every now and then, not sure if there's anything that can be done about that realistically